src/lib.rs

```diff
@@ -3,24 +3,25 @@
 //! version=4 or programatically calling nfs_set_version(nfs, NFS_V4) before
 //! connecting to the server/share.
 //!
-
-#![allow(non_upper_case_globals)]
-
 use libnfs_sys::*;
 use nix::fcntl::OFlag;
 use nix::sys::stat::Mode;
 
-use std::ffi::{CStr, CString, c_char};
+use std::ffi::{CStr, CString};
 use std::io::{Error, ErrorKind, Result};
 use std::mem::zeroed;
+use std::os::raw::{c_int, c_char};
 use std::os::unix::ffi::OsStrExt;
 use std::path::{Path, PathBuf};
 use std::ptr;
-use std::rc::Rc;
+use std::sync::Arc;
 
 #[derive(Clone)]
 struct NfsPtr(*mut nfs_context);
 
+unsafe impl Send for NfsPtr{}
```

Comment:
I think back when I first wrote this I tried to figure out if the nfs pointer is thread safe. Could you point me towards some code or docs?

Reply:
I updated the implementation to enclose nfs_context within a Mutext